extra/3-magic-8-ball.js

```diff
@@ -45,8 +45,41 @@
 
 // This should log "The ball has shaken!"
 // and return the answer.
+
+
+const answers = [["It is certain",
+  "It is decidedly so",
+  "Without a doubt",
+  "Yes - definitely",
+  "You may rely on it"],
+
+["As I see it, yes",
+  "Most likely",
+  "Outlook good",
+  "Yes",
+  "Signs point to yes"],
+
+["Reply hazy, try again",
+  "Ask again later",
+  "Better not tell you now",
+  "Cannot predict now",
+  "Concentrate and ask again"],
+
+["Don't count on it",
+  "My reply is no",
+  "My sources say no",
+  "Outlook not so good",
+  "Very doubtful"]];
+
+
 function shakeBall() {
   //Write your code in here
+
+  let firstItem = Math.floor(Math.random() * 4);
+  let secondItem = Math.floor(Math.random() * 5);
```

Comment:
So `Math.random()` returns a random number _between 0 and 1_ (e.g. `0.6291232`). Therefore, when trying to get random _integers_ (i.e. for indexing an array) we tend to multiply by *array.length*, then get the `floor`. This means that we generate a random number between 0 and `array.length`, then get the `floor` (round down to nearest integer). This way, we generate a random number between `0 - array.length`, then get the `floor`, such that we are guaranteed a random number that is either 0,1,2,3...array.length-1 (remember, the last index of an array is it's length -1 as we start at `0`).

You _could_ avoid hard coding the 4 or 5 here by getting the array lengths, but that's beyond what is expected here

Reply:
Thanks. updated.